### PR TITLE
BUG: BooleanArray.value_counts dropna

### LINKED ISSUE #30685: BooleanArray, StringArray value_counts na_value
```
arr = pd.array([True, False, np.nan])

>>> arr
<BooleanArray>
[True, False, NA]
Length: 3, dtype: boolean

>>> arr.value_counts(dropna=False)
True     1
False    1
False    1
dtype: int64
```

Instead of two False entries, one of them should be pd.NA right?  cc @TomAugspurger 

```
arr = pd.array(['foo', 'bar', None])

>>> arr.value_counts(dropna=False)
foo    1
bar    1
NaN    1
dtype: int64
```

The NaN should be pd.NA right?

Side-note: the defaults for dropna are not consistent across our EAs

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/30685

### CODE REVIEW

**Comment by jbrockmendel** [comment]
LGTM pending green

File: pandas/tests/arrays/test_boolean.py [hunk 1]
```diff
@@ -856,3 +856,10 @@ def test_arrow_roundtrip():
     result = table.to_pandas()
     assert isinstance(result["a"].dtype, pd.BooleanDtype)
     tm.assert_frame_equal(result, df)
+
+
+def test_value_counts_na():
```
Comment (jreback) [inline pandas/tests/arrays/test_boolean.py]: is the result an object dtyped Series when dropna=True? (add a test as well)

Comment (jorisvandenbossche) [reply pandas/tests/arrays/test_boolean.py]: In neither cases (dropna True of False) is the result an object dtype series, it is always integer (it are just counts).

That said, should the result here rather be a nullable integer type? Not that there are nulls here, but in the light of "trying to return nullable types as much as possible from operations involving nullable types".

Comment (jreback) [reply pandas/tests/arrays/test_boolean.py]: hmm, yeah i think we should just move this to return a nullable integer (as this is new api). will promote consistency in the future.

Comment (TomAugspurger) [reply pandas/tests/arrays/test_boolean.py]: Hmm, OK, will update these.

Comment (TomAugspurger) [reply pandas/tests/arrays/test_boolean.py]: And so API breaking change for IntegerARrray.value_counts to return a nullalble int dtype too?

Comment (jorisvandenbossche) [reply pandas/tests/arrays/test_boolean.py]: I would consider doing that, yes.

**Comment by TomAugspurger** [comment]
OK, updated {String,Integer,Boolean}Array.value_counts to return a nullable int64 dtype.

I moved BooleanArray.value_counts and IntegerArray.value_counts to the base class.

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -411,6 +411,25 @@ Use :meth:`arrays.IntegerArray.to_numpy` with an explicit ``na_value`` instead.
...
+
+   >>> pd.Series([2, 1, 1, None], dtype="Int64").value_counts().dtype
+   dtype('int64')
+
+*pandas 1.0.0*
+
+.. ipython:: python
+
+   >>> pd.Series([2, 1, 1, None], dtype="Int64").value_counts().dtype
+   Int64Dtype()
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.0.0.rst]: Don't show the output (or >>> prompt), or otherwise also make it a code-block

**Review by jorisvandenbossche — COMMENTED** [review]
Minor comment, looks good for the rest

File: doc/source/whatsnew/v1.0.0.rst (hunk 2 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v1.0.0.rst]: Whoops, thanks.

File: pandas/tests/arrays/test_boolean.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/arrays/test_boolean.py]: > That said, should the result here rather be a nullable integer type? Not that there are nulls here, but in the light of "trying to return nullable types as much as possible from operations involving nullable types".

I think the "trying to return nullable..." would suggest we would want to keep the dtype for the _index_ here, not the values.  It seems weird that the values would be anything other than np.int64

Comment (TomAugspurger) [reply pandas/tests/arrays/test_boolean.py]: This discussion was about the values, not the index. I think the idea is to keep operations on these nullable types in the nullable type arena.

Comment (jbrockmendel) [reply pandas/tests/arrays/test_boolean.py]: Right, and I'm saying it _doesnt_ make sense to keep the values nullable

Comment (TomAugspurger) [reply pandas/tests/arrays/test_boolean.py]: Can you expand on why you find it weird?

It's true that the result of a value_counts will always have no NAs, but returning a nullable int type prevents a reintroduction of NAs in subsequent operations from converting to float.

Comment (jbrockmendel) [reply pandas/tests/arrays/test_boolean.py]: > Can you expand on why you find it weird?

The motivation is to maintain consistency of "operations with nullable types return nullable types".  But making value_counts().values return IntNA breaks the consistency of "values_counts().values is always np.int64".  So it's a wash on "maintaining consistency".

Ideally we'd retain the dtype in the value_counts().index, and it seems like we're saying here "well we cant do that, so let's shoehorn the dtype into the values"

Comment (TomAugspurger) [reply pandas/tests/arrays/test_boolean.py]: > it seems like we're saying here "well we cant do that, so let's shoehorn the dtype into the values"

No, I don't think we're saying that. I think we're saying we find a nullable integer dtype to be more useful.

Comment (jbrockmendel) [reply pandas/tests/arrays/test_boolean.py]: Not a hill I want to die on.

**Comment by jreback** [comment]
let me clarify my comments

i think the index should be nullable ints as that preserves the intent here
but agree the values should be int64 - this is a count after all

File: pandas/tests/arrays/test_boolean.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/arrays/test_boolean.py]: commented below

**Comment by TomAugspurger** [comment]
Thanks for clarifying.

The index can't be nullable yet. @jorisvandenbossche thoughts?

File: doc/source/whatsnew/v1.0.0.rst [hunk 3]
```diff
@@ -411,6 +411,24 @@ Use :meth:`arrays.IntegerArray.to_numpy` with an explicit ``na_value`` instead.
 
    a.to_numpy(dtype="float", na_value=np.nan)
 
+**value_counts returns a nullable integer dtype**
+
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: actually, even though is a breaking change, it puts nullable in more general usage. so actually i think this is a good change.

**Comment by jorisvandenbossche** [comment]
> Ideally we'd retain the dtype in the value_counts().index, and it seems like we're saying here "well we cant do that, so let's shoehorn the dtype into the values"

IMO, ideally we retain the nullable dtype in *both* the index and the values. But indeed, for index we can't do that yet, so for now it's only the values.

>  It seems weird that the values would be anything other than np.int64

We are still returning an int dtype. But in the "nullable dtype universe"-subsystem, the "int64" dtype is the nullable int64 dtype.

**Comment by TomAugspurger** [comment]
So I think that's some +1s and a -0 (https://github.com/pandas-dev/pandas/pull/30824#discussion_r364862869). Merging in an hour or so unless @jbrockmendel objects.

**Comment by jbrockmendel** [comment]
No objection here.

**Comment by TomAugspurger** [comment]
Thanks.

**Pull request merged**